test-data/unit/check-classvar.test

```diff
@@ -269,6 +269,21 @@ class A(metaclass=ABCMeta):
 class B(A):
     x = 0  # type: ClassVar[int]
 
+[case testOverrideWithPropertyDecorator]
+from typing import ClassVar
+
+class A:
+    x: ClassVar[int]
```

Comment:
Could you add a testcase with `ClassVar[property]`? I suspect that isn't supported even without this PR (#18504) and don't know if it is important enough, but better be explicit about this.

```
class Parent:
   foo: ClassVar[property]

class Child(Parent):
    @property
    def foo(self) -> None: ...
    @foo.setter
    def foo(self, _: None) -> None: ...
```

(this should ideally be accepted, but probably isn't yet)